### PR TITLE
Add tests for external wincmds

### PR DESCRIPTION
**Pull request by coditva**
_The first 4 commits will become obsolete when merging with `grid-split`; remove them then._

Desired tests:
- [x] default screen
- [x] `:split`, `:vertical split`
- [x] `CTRL_W r`
- [x] `CTRL_W x`
- [x] `CTRL_W J`
- [x] `CTRL_W T`
- [x] `CTRL_W =`
- [x] `:resize`, `:vertical resize`
- [x] `CTRL_W +`
- [x] `CTRL_W j`
- [x] `CTRL_W c`

### CODE REVIEW

File: test/functional/ui/ext_win_spec.lua [hunk 1]
```diff
@@ -57,4 +59,44 @@ describe('In ext-win mode', function()
     ]])
   end)
 
+  describe(':split', function()
+    it('sends correct events', function()
```
Comment (coditva) [inline test/functional/ui/ext_win_spec.lua]: @bfredl, should I be testing the events like this one? And if yes, how can I inspect the events?

File: test/functional/ui/ext_win_spec.lua [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+  local nvim_argv = {helpers.nvim_prog, '-u', 'NONE', '-i', 'NONE', '-N',
+                     '--cmd', 'set shortmess+=I background=light noswapfile belloff= noshowcmd noruler',
+                     '--embed'}
+
+  before_each(function()
+    local screen_nvim = spawn(nvim_argv)
+    set_session(screen_nvim)
+    screen = Screen.new(15, 5)
+    -- TODO(utkarshme): should not have to set ext_multigrid
+    screen:attach({ext_windows=true, ext_multigrid=true})
```
Comment (coditva) [inline test/functional/ui/ext_win_spec.lua:18]: @bfredl The tests start failing if I remove `ext_multigrid=true` from this line. Whereas it works in the python-gui without setting it explicitly.

Comment (bfredl) [reply test/functional/ui/ext_win_spec.lua:18]: hmm strange, do you set that `ext_windows` imply `ext_multigrid`, if so, where?

Comment (coditva) [reply test/functional/ui/ext_win_spec.lua:18]: This line: https://github.com/UtkarshMe/neovim/pull/2/files#diff-acdca5aaf6a1d3676bd25ca27959b985R117

Comment (bfredl) [reply test/functional/ui/ext_win_spec.lua:18]: hmm looks correct to me, can look into it more tomorrow if still a problem.

Comment (coditva) [reply test/functional/ui/ext_win_spec.lua:18]: Roger that!

**Pull request merged**